rs/ethereum/ledger-suite-orchestrator/src/state/mod.rs

```diff
@@ -213,7 +215,100 @@ pub type GitCommitHash = Hash<GIT_COMMIT_HASH_LENGTH>;
 
 #[derive(Clone, PartialEq, Debug, Default, Deserialize, Serialize)]
 pub struct ManagedCanisters {
+    /// Canisters for an ERC-20 token
     canisters: BTreeMap<Erc20Token, Canisters>,
+    /// Canisters for non ERC-20 token, identified by their token symbol
+    #[serde(default)]
+    other_canisters: BTreeMap<TokenSymbol, Canisters>,
+}
+
+impl ManagedCanisters {
+    pub fn find_by_id(&self, token_id: &TokenId) -> Option<&Canisters> {
+        match token_id {
+            TokenId::Erc20(contract) => self.canisters.get(contract),
+            TokenId::Other(symbol) => self.other_canisters.get(symbol),
+        }
+    }
+
+    pub fn get_mut(&mut self, token_id: &TokenId) -> Option<&mut Canisters> {
+        match token_id {
+            TokenId::Erc20(contract) => self.canisters.get_mut(contract),
+            TokenId::Other(symbol) => self.other_canisters.get_mut(symbol),
+        }
+    }
+
+    pub fn insert_once(&mut self, token_id: TokenId, canisters: Canisters) {
+        assert_eq!(
```

Comment:
Why is an `assert` preferable to idempotency here (silently doing nothing if the token has been added before)?

Reply:
> silently doing nothing if the token has been added before

I think integrity of state for canisters is generally more important than liveness (stuck on a panic) or usability (single panic like here). It would be a BUG to try to add twice the same token, so better to stop right there and let the user know (in that case the proposer) rather than silence it, which may have undesired side-effects (e.g., the user was expecting a panic but there was none, leading to more problems down the road).